tasks/irsInstances.ts

```diff
@@ -70,7 +75,66 @@ async function writeIrsCreationTransactionsToGnosisSafeTemplate(
   );
   const output = mustache.render(template, data);
 
-  console.log("Output:\n", output);
+  const jsonDir = path.join(__dirname, "JSONs");
+  const outFile = path.join(jsonDir, "createIrsTransactions.json");
+  if (!fs.existsSync(jsonDir)) {
+    fs.mkdirSync(jsonDir);
+  }
+  fs.writeFileSync(outFile, output);
+
+  console.log("Output written to ", outFile.toString());
+}
+
+let builtMapOfRateOracles = false;
+const mapOfRateOracleAddressesToNames = new Map<string, string>();
+
+// gets the verified contract type of a given address from etherscan
+async function getContractName(
+  hre: HardhatRuntimeEnvironment,
+  address: string
+) {
+  if (!builtMapOfRateOracles) {
+    for (const rateOracleType of [
+      "AaveRateOracle",
+      "AaveBorrowRateOracle",
+      "CompoundRateOracle",
+      "CompoundBorrowRateOracle",
+      "LidoRateOracle",
+      "RocketPoolRateOracle",
+    ]) {
+      for (const underlying of [
+        "",
+        "USDC",
+        "USDT",
+        "DAI",
+        "ETH",
+        "WETH",
+        "cDAI",
+        "cUSDT",
+      ]) {
+        const contractName = underlying
+          ? `${rateOracleType}_${underlying}`
+          : rateOracleType;
+        try {
+          const contractAddress = (await hre.ethers.getContract(contractName))
+            .address;
+          mapOfRateOracleAddressesToNames.set(
+            contractAddress.toLowerCase(),
+            contractName
+          );
+        } catch (e) {
+          // Ignore this combination; not all combinations are deployed
+        }
+      }
+    }
+    builtMapOfRateOracles = true;
+  }
+
+  if (mapOfRateOracleAddressesToNames.has(address.toLowerCase())) {
+    return mapOfRateOracleAddressesToNames.get(address.toLowerCase());
+  } else {
+    return "<unknown>";
```

Comment:
Is there any advantage or returning this instead of throwing error?

Reply:
Yeah. It's not really an error cos we have some (abandoned) IRS instances that don't use any of the latest rate oracles.

Reply:
Fair enough.